src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/EndOfServiceReportService.kt

```diff
@@ -34,8 +34,8 @@ class EndOfServiceReportService(
       createdBy = authUserRepository.save(createdByUser),
       createdAt = OffsetDateTime.now(),
     )
-
-    return endOfServiceReportRepository.save(endOfServiceReport)
+    return endOfServiceReportRepository.findByReferralId(referralId)
+      ?: return endOfServiceReportRepository.save(endOfServiceReport)
```

Comment:
can we omit the second return keyword here?

Reply:
Good spot Marco, will do that. Thanks